src/js/chartAuthoring.js

```diff
@@ -151,14 +210,43 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
     // Adds aria attributes that only make sense within the context of the overall chart authoring tool
     // Specifically, adds:
     // - an aria-controls attribute for the dataEntryPanel's form referencing the unique IDs of the pie and legend
+    // - an aria-controls attribute for the reset button referencing the dataEntryPanel form
     floe.chartAuthoring.addAriaConnections = function(that) {
-
         var legendId = fluid.allocateSimpleId(that.chartAuthoringInterface.pieChart.legend.locate("table")),
             pieId = fluid.allocateSimpleId(that.chartAuthoringInterface.pieChart.pie.locate("pie")),
-            totalId = fluid.allocateSimpleId(that.chartAuthoringInterface.dataEntryPanel.locate("totalValue"));
-
+            totalId = fluid.allocateSimpleId(that.chartAuthoringInterface.dataEntryPanel.locate("totalValue")),
+            dataEntryFormId = fluid.allocateSimpleId(that.chartAuthoringInterface.dataEntryPanel.locate("dataEntryForm"));
 
         that.chartAuthoringInterface.dataEntryPanel.locate("dataEntryForm").attr("aria-controls", legendId + " " + pieId + " " + totalId);
+        that.chartAuthoringInterface.locate("reset").attr("aria-controls", dataEntryFormId);
+    };
+
+    // Updates the chart authoring tool's data entry panel to the provided dataset
+    // Passing an empty dataset will clear and reset the form
+    // Triggers change events to trigger model updating and propagation to other
+    // interface elements
+    floe.chartAuthoring.updateDataEntryPanel = function (that, dataSet) {
+        // Clear any existing data entries
+        var dataEntries = that.chartAuthoringInterface.dataEntryPanel.locate("dataEntry");
+
+        var dataEntryLabelSelector = that.chartAuthoringInterface.dataEntryPanel.dataEntry.options.selectors.label;
+        var dataEntryValueSelector = that.chartAuthoringInterface.dataEntryPanel.dataEntry.options.selectors.value;
+
+        dataEntries.each(function() {
+            $(this).find(dataEntryLabelSelector).val("").trigger("change");
```

Comment:
Line 236 and 237 perform the same functionality as line 247, 248. We probably can create a function like: 
```
floe.chartAuthoring.updateDataEntryRow = function (that, index, label, value) {
    var dataEntryComponentName = index === 0 ? "dataEntry" ? "dataEntry-" + index;
    that[dataEntryComponentName].locate("label").val(label).trigger("change");
    that[dataEntryComponentName].locate("value").val(value).trigger("change");
}
```
So we can get rid of the use of ```find()``` to locate fields more accurately as well as reusing this functionality.